### PR TITLE
Added Codecov Workflow

### PR DESCRIPTION
**Pull request by UmairK5669**
**Description:**
Added a Codecov workflow that will allow `pytest` to perform any tests and upload the coverage to codecov using the `pytest-cov` dependency that has been added to `pyproject.toml`. This workflow will run on code that is pushed to the `main` branch as well as PRs that intend to merge into `main`. 

**Related Issues:**
#12 

**Outstanding Sub-Issues:**
This workflow simply uploads the test coverage to Codecov and doesn't check to see if it meets the threshold. 

**Tips for the Reviewer:**
As there are no tests that Ria has presently, I would highly recommend adding dummy tests to test the workflow. After adding dummy tests run `poetry run pytest` to first ensure the tests are being ran, then push the code to this branch and the workflow will run automatically. 

**Checklist:**
Please be sure to check all boxes honestly. This is to ensure a smooth development process, and to reduce the 
likelihood of needing to make additional changes later on.

- [X] I understand that changes authored by individuals who have not signed a Contributor License Agreement (CLA), 
or whose authorship we cannot verify, may not be accepted.
- [X] These contribution address a triaged issue.
- [X] I have performed a self-review of my code and my changes generate no new warnings.
- [X] I have included unit tests for my code contributions, and all tests are passing.
- [X] The docstrings are complete, include doctests demonstrating usage, and are properly formatted. 
I have verified that any updates to the project documentation are complete and look okay.
- [X] I have confirmed all my code contributions are formatted in accordance with the project's Flake8 style 
configuration.
- [X] No unnecessary changes have been made to the Poetry lock file, but `pyproject.toml` and the Poetry lock file 
have been updated to reflect any dependency changes.
- [X] I acknowledge that upon the submission of this pull request, Qoherent will assume the copyright for this 
contribution.

A heartfelt thank you from everyone at Qoherent and the broader radio community for taking the time to contribute to 
RIA Core. 🙏💖

### CODE REVIEW

File: pyproject.toml
Comment (UmairK5669) [inline pyproject.toml:1]: Should these dependencies be moved to the `tool.poetry.group.test.dependencies`?

**Comment by UmairK5669** [comment]
Looks good overall, should we also remove the dummy tests or keep them in until more tests get added to RIA?

**Comment by UmairK5669** [comment]
Note: We will need to squash the commits at merge time.

**Comment by UmairK5669** [comment]
> It looks like test_dummy.py is already in main, should we be removing it?

Yeah, I think they should, they don't really serve a purpose and were only for testing this workflow.

**Review by fordg1 — APPROVED** [review]

**Pull request merged**